base/applications/utilman/precomp.h

```diff
@@ -50,6 +50,43 @@ typedef struct _UTILMAN_STATE
     BOOL    bState;
 } UTILMAN_STATE, *PUTILMAN_STATE;
 
+typedef struct _REGISTRY_SETTINGS
+{
+    /* Accessibility Registry settings */
+    LPCWSTR wszAppPath;
+    DWORD dwAppType;
+    DWORD dwClientControlCode;
+    LPCWSTR wszAppName;
+    LPCWSTR wszErrorOnLaunch;
+    BOOL bHideClient;
+    BOOL bStartWithUtilman;
+    BOOL bStartWithROS;
+    LPCWSTR wszHungRespondAction;
+    DWORD dwHungTimeOut;
+
+    /* Utility Manager Registry settings */
+    BOOL bShowWarning;
+} REGISTRY_SETTINGS, *PREGISTRY_SETTINGS;
+
+typedef struct _REGISTRY_DATA
+{
+    /* On-Screen Keyboard Registry data */
+    LPCWSTR lpwsOskPath;
```

Comment:
Use buffers instead of pointers.

Reply:
Care to explain?

Reply:
Because when you'll need to retrieve their data from the registry you'll have to allocated a buffer from the heap, so katahiromz thought it would be simpler here to use a fixed-sized buffer instead. I think the heap allocation, while technically more cumbersome, would be better as it would allow you to avoid all these hard-coded buffer sizes.